src/Orchard.Web/Modules/Orchard.Redis/Services/RedisCacheNotificationProvider.cs

```diff
@@ -0,0 +1,35 @@
+﻿using Orchard.Environment.Features;
+using Orchard.Localization;
+using Orchard.UI.Admin.Notification;
+using Orchard.UI.Notify;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Orchard.Redis.Configuration {
+    public class RedisCacheNotificationProvider : INotificationProvider {
+        private readonly RedisConnectionProvider _redisConnectionProvider;
+        private readonly FeatureManager _featureManager;
+
+        public RedisCacheNotificationProvider(FeatureManager featureManager, RedisConnectionProvider redisConnectionProvider) {
+            _redisConnectionProvider = redisConnectionProvider;
+            _featureManager = featureManager;
+            T = NullLocalizer.Instance;
+        }
+
+        public Localizer T { get; set; }
+
+        public IEnumerable<NotifyEntry> GetNotifications() {
+            //verify if module is enabled first
+            if (!_featureManager.GetEnabledFeatures().Where(f => f.Id == "Orchard.Redis.Caching").Any()) {
```

Comment:
Wrong, you just need [OrchardFeature("Orchard.Redis.Caching")] on the class. Look at the service for the same feature for an example.